리뷰

nodonghui opened the pull request.

1m1nkim committed:
Feat: 도서 목록전체 조회, 도서 샘플 데이터 추가

rOyalFruit committed:
Feat: 샘플 데이터 생성, 기능 추가

- 유저, 리뷰, 도서 샘플 데이터 생성
- 전체 도서, 특정 도서 조회
- 전체 리뷰, 특정 도서 리뷰 조회

rOyalFruit committed:
Feat: 由щ럭 ��, ��� 湲곕 異�

rOyalFruit committed:
Feat: �λ�援щ 異�, �� 湲곕 援ы�

rOyalFruit committed:
Feat: 장바구니 삭제 기능 추가

Commit:
feat: 주문 조회 기능

rOyalFruit committed:
Feat: 비회원 상태에서 담아놓은 장바구니 항목을 로그인 시 DB에 저장

rOyalFruit committed:
Feat: 由щ럭 �¦ 湲곕 異�

1m1nkim committed:
Feat: 메인화면, 장바구니, 검색화면, 책 상세 페이지 Ui 구성

Commit:
feat: 주문 상세 조회 생성

rOyalFruit committed:
Refactor: getAllBooks() 硫¦蕤� ��

- 諛� �¢��¦ List<Book>�¦  Page<Book>�쇰� 蹂¢寃
- �竈吏¢�ㅼ竈� 湲곕 異�
- �젹 湲곗� 異�

Commit:
feat: 주문 삭제 생성

Commit:
refactor: 주문삭제 리팩토링

rOyalFruit committed:
Feat: 특정 도서에 달린 리뷰 개수 갱신

- 댓글 작성 시 리뷰 개수 증가
- 댓글 삭제 시 리뷰 개수 감소

rOyalFruit committed:
Refactor: getAllReviews(), getReviewsById() 메서드 수정

- 반환 타입을 List<Review>에서  Page<Review>로 변경
- 페이지네이션 기능 추가
- 정렬 기준 추가

Commit:
feat: 전역 예외 처리 클래스 추가

rOyalFruit committed:
Refactor, Feat: 장바구니 추가 메서드 수정 및 조회 기능 추가

- 장바구니 추가 메서드 리팩터링
  - updateCartItems() 메서드 재사용하여 중복 코드 제거
- 장바구니 조회 기능 구현

Commit:
fix: 주문 삭제 후 조회 시 존재하지 않는 주문에 대한 예외 처리

rOyalFruit committed:
Feat: Book 엔티티 필드 추가 및 이름 수정

rOyalFruit committed:
Feat: 리뷰 등록, 수정, 삭제 시 해당 도서의 평점에 반영

- Book 엔티티의 rating에는 평점 합계 저장
- 프론트에서 (총점/리뷰 개수)로 평균 평점을 반환

Commit:
feat: CORS 설정 추가 - localhost:3000에서의 요청 허용

1m1nkim committed:
Feat: 도서 가져오기

-백엔드 데이터 연결, api.js, webconfig 생성

rOyalFruit committed:
Feat: 메인 페이지 작성

- 백엔드 eventBanner 폴더에 있는 이미지들을 프론트 이벤트 배너에 보여주도록 함
- 이벤트 배너의 < > 버튼을 컴포넌트로 분리

BeigeBearDev committed:
feat: build.gradle에 dependencies 추가

rOyalFruit committed:
Merge remote-tracking branch 'origin/Feature1-Book,Cart' into Feature1-Book,Cart

# Conflicts:
#	backend/src/main/java/com/ll/nbe342team8/global/config/WebConfig.java
#	frontend/package-lock.json
#	frontend/package.json

nodonghui committed:
fect: 마이페이지, 배송 정보 등록 기능 1차 완성

jwt를 이용한 인가 기능은 구현하지 않았습니다.
추가적인 업데이트가 필요합니다.

nodonghui committed:
fix: .gitignore 규칙 갱신

rOyalFruit committed:
Fix: 메인 페이지에서 유효하지 않은 이미지 URL 처리 로직 개선

- 각 도서별 이미지 에러 상태 개별 관리를 위해 imageErrors 객체 도입
- URL 유효성 검사 함수(isValidImageUrl) 추가
- 이미지 표시 조건 강화: URL 존재 여부, 유효성, 로드 성공 여부 확인
- 유효하지 않은 이미지 URL이나 로드 실패 시 NoImage 컴포넌트로 대체

1m1nkim committed:
Feat: 장바구니 조회, 수정, ui 개발

- 추후 memberId에 따른 변경 필요

nodonghui committed:
refactor : 코드 리팩토링

dto 부분에 유효성 검사를 추가하고
간단한 테스트 케이스를 일부를 추가했습니다.

BeigeBearDev committed:
feat: 관리자 상품 등록 기능
- 알라딘 외부 API 호출
- WebClient 적용

1m1nkim committed:
Feat: 장바구니 검색 기능 추가

- 추후 Paging 처리 후 반환 예정

rOyalFruit committed:
Feat: 도서 상세 페이지 작성

- 도서 상세 정보 보여주기
- 리뷰 보여주기(페이징, 정렬 기능 구현)
- 리뷰 작성
- 교환/반품 탭에 테이블 추가

BeigeBearDev committed:
feat: 관리자 상품 수정 기능
- PATCH 요청
- RequestDto에서 유효성 검증

rOyalFruit committed:
Merge remote-tracking branch 'origin/Feature1-Book,Cart' into Feature1-Book,Cart

nodonghui committed:
refactor: dto 수정

boolean에 @notblank를 notnull로 바꿈
deliveryInformation에 이스케이프변환 적용
deliveryInformationDto의 특수문자 제외 확인 어노테이션을 제거했습니다.

BeigeBearDev committed:
Merge pull request #6 from prgrms-be-devcourse/feature-admin

feat: 상품 등록 기능 추가

rOyalFruit committed:
Merge remote-tracking branch 'origin/dev' into Feature1-Book,Cart

# Conflicts:
#	backend/build.gradle
#	backend/src/main/java/com/ll/nbe342team8/domain/book/book/controller/BookController.java
#	backend/src/main/java/com/ll/nbe342team8/domain/book/book/dto/BookResponseDto.java
#	backend/src/main/java/com/ll/nbe342team8/domain/book/book/entity/Book.java
#	backend/src/main/java/com/ll/nbe342team8/domain/book/book/repository/BookRepository.java
#	backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/BookService.java
#	backend/src/main/java/com/ll/nbe342team8/domain/book/category/entity/Category.java

nodonghui committed:
Resolve merge conflict with feature-order-search

rOyalFruit committed:
Fix: 브랜치 병합 후 생긴 에러 수정

rOyalFruit committed:
Merge pull request #8 from prgrms-be-devcourse/Feature1-Book,Cart

Feature1 book,cart

nodonghui committed:
Resolve merge conflicts with dev

nodonghui committed:
merge process

nodonghui committed:
Merge pull request #11 from prgrms-be-devcourse/temp-merge-nodonghui

Temp merge nodonghui

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/controller/BookController.java:
```diff
@@ -1,7 +1,73 @@
+@RequestMapping("/books")
+@Tag(name = "Book", description = "Book API")
+@RequiredArgsConstructor
 public class BookController {
```
- 전반적으로 입력 받는 Parameter에 대한 검증이 없네요. (예를 들어, getAllBooks의 pageSize 값으로 -100을 넘기면 어떻게 될까요?)

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/controller/BookController.java:
```diff
@@ -1,7 +1,73 @@
+    }
+
+    @Operation(summary = "특정 도서 조회")
+    @GetMapping("/{book-id}")
```
얘만 캐멀케이스를 안 쓰는 이유가 있을까요?
이거 때문에 `@PathVariable` 어노테이션 뒤에 불필요한 속성값이 같이 붙어버렸네요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/controller/BookController.java:
```diff
@@ -1,7 +1,73 @@
+
+    @PostMapping("/admin/books")
+    public ResponseEntity<?> addBook(@RequestParam(required = false) String isbn13) {
+        if (isbn13 == null) {
```
그냥 required=false 옵션을 빼면 굳이 분기처리를 안해도 되지 않을까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/controller/BookController.java:
```diff
@@ -1,7 +1,73 @@
+    }
+
+    @PatchMapping("/admin/books/{bookId}")
+    public ResponseEntity<BookResponseDto> updateBookPart(@PathVariable("bookId") Long bookId,
```
`@PathVariable Long bookId`

Account-4-Review replied:
위 메서드에서는 `long` 을 쓰고, 여기서는 `Long` 을 쓰는데 혹시 둘을 구분하는 내부적인 컨벤션이 존재하는걸까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/dto/BookPatchRequestDto.java:
```diff
@@ -0,0 +1,50 @@
+    private String title;
+    private String author;
+
+    @Pattern(regexp = "^[0-9]{10}$", message = "ISBN은 10자리 숫자여야 합니다.")
```
- Controller 에서 `@Valid` 를 안 쓰고 있어서, Validation 이 수행되지 않을 것으로 보여요.
- Dto는 가변일 필요가 없고, (아마 이 텍스트를 보고 제가 말하겠지만) 모던 자바로 넘어가면서 데이터의 불변성이 아주 중요한 이슈로 떠오른 상황이에요.`@Data` 대신, Record로 대체해 주세요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/dto/BookPatchRequestDto.java:
```diff
@@ -0,0 +1,50 @@
+    private Integer status;
+
+    @DecimalMin(value = "0.0", message = "평점은 0 이상이어야 합니다.")
+    private Float rating;
```
Float은 정확도 문제로 권장되지 않습니다. Double로 대체해 주세요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/dto/BookPatchRequestDto.java:
```diff
@@ -0,0 +1,50 @@
+        return status == null || status == 0 || status == 1;
+    }
+
+    private Integer status;
```
status를 숫자가 아닌 Enum 으로 관리할 순 없을까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/dto/BookResponseDto.java:
```diff
@@ -0,0 +1,56 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+public record BookResponseDto(Long id,
```
어떤 곳에서는 record를 쓰고, 어떤 곳에서는 @Data 를 쓰는걸 보니, 팀 내 컨벤션이 정확하게 확정되지 않은 것 같아요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/entity/Book.java:
```diff
@@ -1,40 +1,108 @@
+    @Column(name = "isbn")
+    private String isbn;       // ISBN
+
+    @NotNull
```
위에서는 `@Column` 을 쓰는데, 왜 여기부터는 없을까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/entity/Book.java:
```diff
@@ -1,40 +1,108 @@
+    }
+
+    public void update(BookPatchRequestDto requestDto) {
+        if (requestDto.getTitle() != null) this.title = requestDto.getTitle();
```
(좀 챌린징한 부분이에요.)
일일히 반복문 안 쓰고 update 할 수 있을까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/review/entity/Review.java:
```diff
@@ -1,23 +0,0 @@
-@AllArgsConstructor
-public class Review extends BaseTime {
-
-
```
요 공백들도 컨벤션에 맞게 정리해주세요 ㅎㅎ

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/BookService.java:
```diff
@@ -1,9 +1,113 @@
+        List<Sort.Order> sorts = new ArrayList<>();
+        sorts.add(sortType.getOrder());
+
+        Pageable pageable = PageRequest.of(page, pageSize, Sort.by(sorts));
```
그냥 `sortType.getOrder` 로도 대체가 되겠네요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/BookService.java:
```diff
@@ -1,9 +1,113 @@
+        sorts.add(sortType.getOrder());
+
+        Pageable pageable = PageRequest.of(page, pageSize, Sort.by(sorts));
+        return bookRepository.findAll(pageable);
```
모든 로직 전체에 pageSize의 제한이 없는데요.
그렇다면 만약 제가 pageSize=10000000 을 넣으면 어떻게 될까요?

Account-4-Review replied:
+ (약간 챌린징)
Spring 의 Page 는 실제 DB 쿼리에서 어떻게 되는지 아시나요? 그것이 어떻게 동작할까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/BookService.java:
```diff
@@ -1,9 +1,113 @@
+    }
+
+    public Book getBookById(Long id) {
+        if (id == null) {
```
애초에 입력 값에서 id가 notnull 임을 validate 하는게 맞지 않을까요?
기본적인 validation 을 service 레이어에서 수행하게 하지 말아주세요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/global/globalExceptionHandler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,45 @@
+package com.ll.nbe342team8.global.globalExceptionHandler;
```
패키지 룰에 맞지 않습니다.
(`Rename this package name to match the regular expression '^[a-z_]+(\.[a-z_][a-z0-9_]*)*$'.`)

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/order/order/exception/GlobalRestExceptionHandler.java:
```diff
@@ -0,0 +1,22 @@
+package com.ll.nbe342team8.domain.order.order.exception;
```
"Global" 이라면서, 왜 order 안에 있나요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/order/order/exception/GlobalRestExceptionHandler.java:
```diff
@@ -0,0 +1,22 @@
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<String> handleIllegalArgumentException(IllegalArgumentException ex) {
+        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ex.getMessage());
```
`IllegalArgumentException` 이 반드시 NOT_FOUND 임이 보장되나요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/global/globalExceptionHandler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,45 @@
+
+
+    @ExceptionHandler(NoSuchElementException.class)
+    public ResponseEntity<?> handle(NoSuchElementException ex) {
```
빈 리스트에서 `getFirst` 를 호출해도 `NoSuchElementException` 이 발생한다는거 아시나요?
이 예외를 이렇게 받는건 신뢰할 수 없는 결과를 낼 것 같아요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/global/globalExceptionHandler/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+
+    @ExceptionHandler(ServiceException.class)
```
커스텀 예외를 정했다면, **절대 비즈니스 로직 내부에 자바 기본 예외를 리턴하지 말아주세요.**
위에서도 지적했듯이 자바 기본 예외는 자바 내부 메서드 호출 과정에서도 발생합니다.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/ExternalBookApiService.java:
```diff
@@ -0,0 +1,38 @@
+            throw new IllegalArgumentException("ISBN13 값은 필수입니다.");
+        }
+
+        String url = "http://www.aladin.co.kr/ttb/api/ItemLookUp.aspx?"
```
UriComponentBuilder 사용

Account-4-Review replied:
```java
        var url = UriComponentsBuilder.fromUriString("/ItemLookUp.aspx")
                .queryParam("ttbkey", ttbkey)
                .queryParam("itemIdType", "ISBN13")
                .queryParam("ItemId", isbn13)
                .queryParam("output", "js")
                .queryParam("Cover", "Big")
                .build()
                .toUriString();
```

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/book/service/ExternalBookApiService.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    public ExternalBookDto searchBook(String isbn13) {
+        if (isbn13 == null || isbn13.isEmpty()) {
```
Apache Commons StringUtils 사용을 해 보세요. 
`StringUtils.isEmpty(isbn13)`

nodonghui committed:
fix: 병합 문제 해결

cartcontroller 문제 발생 -> member 엔티티와 cart 엔티티 사이 관계추가
mysql 사용자와 h2-mode=mysql 사용자간 yml 파일 설정 충돌 발생 -> yml파일 분리
baseInit의 dataInitializer과 baseInitData 충돌 -> baseInitData를 따라가고 baseInitData에 order과 detailOrder추가 필요
현재 두 파일 모두 @Configuration 주석처리해놈

nodonghui committed:
fix : yml파일 민감한 정보 삭제제
ttb key와 db 비밀번호를 삭제하였습니다.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/category/entity/Category.java:
```diff
@@ -1,18 +1,47 @@
+    private String depth5;
+
+    @OneToMany(mappedBy = "categoryId", cascade = CascadeType.ALL)
+    private List<Book> books = new ArrayList<>();
```
해당 카테고리에 책이 한 6만건 있다고 해 봅시다. 그러면 카테고리를 삭제하게 되면 데이터 6만건이 같이 삭제 될텐데요. 좀 불안정하지 않을까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/global/jpa/entity/BaseEntity.java:
```diff
@@ -16,7 +17,7 @@
 public class BaseEntity {
```
SonarQube의 경고문을 그대로 옮겨드릴게요.

`Using @EqualsAndHashCode for JPA entities is not recommended. It can cause severe performance and memory consumption issues. `

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/global/jpa/entity/BaseEntity.java:
```diff
@@ -16,7 +17,7 @@
 
     @Id
-    @GeneratedValue(strategy = IDENTITY) // AUTO_INCREMENT
+    @GeneratedValue(strategy = GenerationType.IDENTITY) // AUTO_INCREMENT
```
모든 key가 전부 AUTO_INCREMENT 라고 보장할 수 있나요?
혹시 특정 key는 그렇지 않아야 한다면 어떡하죠?

id를 baseEntity로 두진 않았으면 합니다.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/review/controller/ReviewController.java:
```diff
@@ -0,0 +1,81 @@
+
+    @PostMapping("/{book-id}/{member-id}")
+    @Operation(summary = "리뷰 등록")
+    public void createReview(@PathVariable("book-id") Long bookId,
```
모든 Response Code가 200인가요?
HttpStatus 에 대해 자세히 알아보고, 필요한 StatusCode를 추가해 주세요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/review/controller/ReviewController.java:
```diff
@@ -0,0 +1,81 @@
+        Book book = bookService.getBookById(bookId);
+        Member member = memberService.getMemberById(memberId);
+
+        Review review = Review.builder()
```
Controller가 되었던 Service가 되었던, Builder 로직을 외부에서 사용하지 않았으면 좋겠습니다.
Builder가 실수의 여지를 줄여주긴 하지만, 코드의 길이를 세로로 늘리는 단점이 있고, Builder를 세팅하는 로직 때문에 다른 로직의 가독성이 나빠지는 단점이 있어요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/book/review/type/SortType.java:
```diff
@@ -0,0 +1,22 @@
+
+import org.springframework.data.domain.Sort;
+
+public enum SortType {
```
비슷한 코드가 있는 것 같은데요, 묶을 순 없을까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/event/controller/EventController.java:
```diff
@@ -0,0 +1,31 @@
+    public List<String> getBannerImages() throws IOException {
+        String location = "classpath:static/images/eventBanner/";
+        PathMatchingResourcePatternResolver resolver = new PathMatchingResourcePatternResolver();
+        Resource[] resources = resolver.getResources(location + "**");
```
요청이 들어올 때 마다 resource 를 뒤지는 것 같은데, 첫 로드 시 한 번만 목록을 뽑을 순 없나요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/member/deliveryInformation/controller/DeliveryInformationController.java:
```diff
@@ -1,9 +1,105 @@
+        Optional<Member> optionalMember = memberService.findByEmail(email);
+
+        //이메일에 대응하는 사용자가 없는 경우 에러 발생
+        if(optionalMember.isEmpty()) { throw new ServiceException(404,"사용자를 찾을 수 없습니다.");}
```
Code Formatting 이 좀 아쉬운데요,
Convention 적용이 안 되었을까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/member/deliveryInformation/dto/DeliveryInformationDto.java:
```diff
@@ -1,13 +1,49 @@
+@NoArgsConstructor
 public class DeliveryInformationDto {
+
+    @JsonProperty("id")
```
이름이 달라야 `@JsonProperty` 가 의미가 있을 것 같은데, 여긴 죄다 똑같네요.

Account-4-Review replied:
이 코드만 스타일이 다른게 눈에 띄는데요, 여러분들끼리 상호 리뷰 시 이런 부분도 체크해주시면 좋겠어요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/standard/util/Ut.java:
```diff
@@ -2,7 +2,7 @@
+import org.apache.commons.text.StringEscapeUtils;
 
 
 public class Ut {
```
클래스 네이밍이 매우 이해하기 어려워보이고요, 하단 static class는 네이밍 컨벤션도 안 맞네요.
다른 팀 코드리뷰에도 본 코드인데, 혹시 어디서 난 코드일까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/member/deliveryInformation/entity/DeliveryInformation.java:
```diff
@@ -32,7 +33,24 @@ public class DeliveryInformation extends BaseTime {
-
+    public DeliveryInformation(ReqDeliveryInformationDto dto, Member member) {
+        this.addressName = Ut.XSSSanitizer.sanitize(dto.getAddressName());
+        this.postCode = Ut.XSSSanitizer.sanitize(dto.getPostCode());
```
여기에만 해당 로직이 추가된 이유가 있을까요?

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/order/order/service/OrderService.java:
```diff
@@ -1,7 +1,62 @@
+        this.memberRepository = memberRepository;
+    }
+
+    public List<OrderDTO> getOrdersByMemberId(Long memberId) {
```
조회시에도 readOnly option 으로 transactional 을 걸어주세요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/order/order/service/OrderService.java:
```diff
@@ -1,7 +1,62 @@
+
+    public List<OrderDTO> getOrdersByMemberId(Long memberId) {
+        // 회원이 존재하는지 먼저 체크
+        Member member = memberRepository.findById(memberId)
```
어디에서나 나올 수 있는 로직이라, 이건 별도의 메서드로 빼서 MemberService 쪽에서 가져갔으면 하네요.

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/order/order/service/OrderService.java:
```diff
@@ -1,7 +1,62 @@
+                .map(order -> new OrderDTO(order.getMember().getId(),
+                        order.getOrderStatus().name(),
+                        order.getTotalPrice()))
+                .collect(Collectors.toList());
```
.toList

Account-4-Review commented on backend/src/main/java/com/ll/nbe342team8/domain/member/member/service/MemberService.java:
```diff
@@ -1,9 +1,45 @@
+    @Transactional
+    public void modifyMember(Member member, PutReqMemberMyPageDto dto) {
+        //사용자 개체 데이터 갱신
+        member.updateMemberInfo(dto);
```
제가 특정 데이터를 수정하고 싶지 않아서 null 을 던지면, 그걸로 덮어씌워지나요?
